src/Tribe/Commerce/Currency.php

```diff
@@ -580,4 +580,256 @@ public function get_symbols_for_codes( $codes ) {
+			 * Filter the Amount of Decimals for EDD
+			 *
+			 * @since TBD
+			 *
+			 * @param int the default number of decimals
+			 */
+			$decimals = apply_filters( 'tribe_edd_format_amount_decimals', 2 );
+
+			return $decimals;
+		}
+
+		return $this->get_currency_number_of_decimals();
+	}
+
+	/**
+	 * Get the Default Amount of Decimals
+	 *
+	 * @since TBD
+	 *
+	 * @return int the amount of decimals
+	 */
+	public function get_currency_number_of_decimals() {
+
+		/**
+		 * Filter the Amount of Decimals
+		 *
+		 * @since TBD
+		 *
+		 * @param int the default number of decimals
+		 */
+		$decimals = apply_filters( 'tribe_format_amount_decimals', 2 );
+
+		return $decimals;
+	}
+
+	/**
+	 * Get the Currency Configuration for all Passed Providers
+	 *
+	 * @since TBD
+	 *
+	 * @param string $provider The ticket provider class name
+	 * @param int    $post_id  The id of the post with tickets
+	 *
+	 * @return array
+	 */
+	public function get_currency_config_for_provider( $providers, $post_id ) {
+
+		if ( ! is_array( $providers ) ) {
+			$providers[] = $providers;
+		}
+
+		$currency = [];
+
+		foreach ( $providers as $provider ) {
+			$currency[ $provider ] = $this->get_currency_by_provider( $post_id, $provider );
+		}
+
+		return $currency;
+	}
+
+	/**
+	 * Get the Currency Formatting Information for a Provider
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $post_id  The id of the post with tickets
+	 * @param string|null $provider The ticket provider class name
+	 *
+	 * @return array an array of formatting details
+	 */
+	public function get_currency_by_provider( $post_id, $provider = null ) {
+		return [
+			'symbol'             => $this->get_provider_symbol( $provider ),
+			'placement'          => $this->get_provider_symbol_position( $provider, $post_id ),
+			'decimal_point'      => $this->get_currency_decimal_point( $provider ),
+			'thousands_sep'      => $this->get_currency_thousands_sep( $provider ),
+			'number_of_decimals' => $this->get_number_of_decimals( $provider ),
+		];
+	}
+
+	/**
+	 * Get Formatted Currency According to a Provider
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $amount   the amount to format
+	 * @param int         $post_id  The id of the post with tickets
+	 * @param string|null $provider The ticket provider class name
+	 *
+	 * @return mixed|void
+	 */
+	public function get_formatted_currency( $amount, $post_id, $provider = null ) {
+
+		$currency = $this->get_currency_by_provider( $post_id, $provider );
+
+		// Format the amount
+		if ( $currency['decimal_point'] == ',' && false !== ( $sep_found = strpos( $amount, $currency['decimal_point'] ) ) ) {
+			$whole  = substr( $amount, 0, $sep_found );
+			$part   = substr( $amount, $sep_found + 1, ( strlen( $amount ) - 1 ) );
+			$amount = $whole . '.' . $part;
+		}
+
+		// Strip , from the amount (if set as the thousands separator)
+		if ( $currency['thousands_sep'] == ',' && false !== ( $found = strpos( $amount, $currency['thousands_sep'] ) ) ) {
```

Comment:
Since we've changed the decimal `,` to `.` - should we just remove them all no matter what?